octodns/provider/constellix.py

```diff
@@ -85,7 +85,7 @@ def _request(self, method, path, params=None, data=None):
 
     @property
     def domains(self):
-        if self._domains is None:
+        if not self._domains:
```

Comment:
Just to double check. I believe this fall into the block any time this is called and there are no zones returned from the API. That probably won't hurt anything since the end result will be the same, but it could result in multiple calls to the API to fetch the empty zone list. Seems OK, but want to make sure that's expected.

Reply:
Yep you were right, after I ran `--debug` I saw a bunch of API calls that weren't necessary. Fixed it up to be cleaner by checking if the ID was found or not.